backend/mobilityAI/Controllers/SensorDataController.cs

```diff
@@ -215,15 +248,69 @@ private List<string> readData(IFormFile fileName)
         /// </summary>
         /// <param name="macAddress">
         /// The MacAddress of the device to retrieve the information
+        /// </param>
         [HttpGet("GetDeviceInfo", Name = "GetDeviceInfo")]
-        public JsonResult GetDeviceInfo(string macAddress) {
+        public JsonResult GetDeviceInfo(string macAddress)
+        {
             var data = from a in _context.Devices
                        join b in _context.Users on a.UserID equals b.Id
                        where (a.Id == macAddress)
-                       select new {a.FriendlyName, b.Name, a.LastSync};
+                       select new { a.FriendlyName, b.Name, a.LastSync };
 
             return new JsonResult(data);
-            
+        }
+
+        /// <summary>
+        /// Allows the machine learning server to send back the labeled data
+        /// </summary>
+        /// <param name="Id">Callback id that's created from the server</param>
+        /// <param name="Activities">File containing timestamps and the activities performed</param>
+        [HttpPost("MlCallback", Name = "MlCallback")]
+        public IActionResult MlCallback(string Id, IFormFile Activities)
+        {
+            if (mlCallbackIds.ContainsKey(Id))
+            {
+                string deviceId;
+                if (mlCallbackIds.TryGetValue(Id, out deviceId))
```

Comment:
How do you handle cases where the server restarts before receiving a callback?

Reply:
The `Id` won't exist and the callback will fail, do we need to be resilient to this scenario?

Reply:
Not sure, you can decide